rbd/migration.go

```diff
@@ -0,0 +1,219 @@
+	MigrationImageError = MigrationImageState(C.RBD_IMAGE_MIGRATION_STATE_ERROR)
+	// MigrationImagePreparing is the representation of
+	// RBD_IMAGE_MIGRATION_STATE_PREPARING from librbd.
+	MigrationImagePreparing = MigrationImageState(C.RBD_IMAGE_MIGRATION_STATE_PREPARING)
+	// MigrationImagePrepared is the representation of
+	// RBD_IMAGE_MIGRATION_STATE_PREPARED from librbd.
+	MigrationImagePrepared = MigrationImageState(C.RBD_IMAGE_MIGRATION_STATE_PREPARED)
+	// MigrationImageExecuting is the representation of
+	// RBD_IMAGE_MIGRATION_STATE_EXECUTING from librbd.
+	MigrationImageExecuting = MigrationImageState(C.RBD_IMAGE_MIGRATION_STATE_EXECUTING)
+	// MigrationImageExecuted is the representation of
+	// RBD_IMAGE_MIGRATION_STATE_EXECUTED from librbd.
+	MigrationImageExecuted = MigrationImageState(C.RBD_IMAGE_MIGRATION_STATE_EXECUTED)
+	// MigrationImageAborting is the representation of
+	// RBD_IMAGE_MIGRATION_STATE_ABORTING from librbd.
+	MigrationImageAborting = MigrationImageState(C.RBD_IMAGE_MIGRATION_STATE_ABORTING)
+)
+
+// MigrationImageInfo provides information about the
+// live migration progress of an image
+type MigrationImageInfo struct {
+	SourcePoolID        int
+	SourcePoolNamespace string
+	SourceImageName     string
+	SourceImageID       string
+	DestPoolID          int
+	DestPoolNamespace   string
+	DestImageName       string
+	DestImageID         string
+	State               MigrationImageState
+	StateDescription    string
+}
+
+// MigrationPrepare prepares a migration
+// creating a target image with a link
+// to source and making source read-only
+//
+// Implements:
+//  int rbd_migration_prepare(rados_ioctx_t ioctx,
+//                            const char *image_name,
+// 	                          rados_ioctx_t dest_ioctx,
+//                            const char *dest_image_name,
+//                            rbd_image_options_t opts);
+//
+func MigrationPrepare(ioctx *rados.IOContext, destIoctx *rados.IOContext, sourceImageName string, destImageName string, rio *ImageOptions) error {
+	cSourceImageName := C.CString(sourceImageName)
+	cDestImageName := C.CString(destImageName)
+	defer func() {
+		C.free(unsafe.Pointer(cSourceImageName))
+		C.free(unsafe.Pointer(cDestImageName))
+	}()
+
+	ret := C.rbd_migration_prepare(
+		cephIoctx(ioctx),
+		cSourceImageName,
+		cephIoctx(destIoctx),
+		cDestImageName,
+		C.rbd_image_options_t(rio.options))
+
+	return getError(ret)
+}
+
+// MigrationPrepareImport prepares a migration for import
+// from a specified source to a new target image
+//
+// Implements:
+//  int rbd_migration_prepare_import(const char *source_spec,
+//	                                 rados_ioctx_t dest_ioctx,
+//                                   const char *dest_image_name,
+//	                                 rbd_image_options_t opts);
+//
+func MigrationPrepareImport(ioctx *rados.IOContext, sourceSpec string, destImageName string, rio *ImageOptions) error {
```

Comment:
See above. Let's match the order of args in the C apis.

Reply:
done